### DISCUSSION TITLE
Will using the flags -S -no-ff during merge do what I think

### DISCUSSION THREAD

**Discussion by jaytelford** [community/community#22940] [New to GitHub]
<p>Hi,</p>
<p>I was wondering if you guys could tell me if I could do this while merging branches.</p>
<p>git merge -S -no-ff branch name</p>
<p>I think the above command should do this:</p>
<p>1: begin the merge process<br>
2: sign the merge commit message with my gpg key<br>
3: keep all commit messages made to branch being merged prior to merge</p>
<p>Will it do what I think it will.do?</p>
<p>Cheers<br>
Jay</p>

**Comment by lee-dohm** [comment, marked as answer]
<p>I believe item 3 is more along the lines of “will create a merge commit even if the commits on the target branch are all descended from the commit pointed to by HEAD”. From <a href="https://git-scm.com/docs/git-merge" rel="nofollow noopener">the documentation</a>:</p>
<blockquote>
<p>Create a merge commit even when the merge resolves as a fast-forward.</p>
</blockquote>
<p>The <code>git merge</code> command is designed to merge two branches back into one, so it typically shouldn’t have anything to do with commit messages (other than the one on the merge commit, if any).</p>

**Comment by jaytelford** [comment]
<p>that makes sense actaually.  I just got a little confused about maintaing commit messages for each file in a certain branch after said branch had been merged back into the default branch (or working branch as they case me be).</p>
